tests/tests/new_proc_macros.rs

```diff
@@ -47,6 +48,16 @@ mod rpc_impl {
 
 		#[subscription(name = "echo", unsub = "no_more_echo", item = u32)]
 		fn sub_with_params(&self, val: u32);
+
+		#[method(name = "params")]
+		fn params(&self, a: u8, b: &str) -> String {
```

Comment:
I reckon it'd be nice to have some more tests to exercise this lisetime stuff (if that makes sense); like, trying a couple of custom types with generics (elided or not) and a type with more than one lifetime to check that we hit a compile error (and preferably a nice one).

Reply:
Added a bunch of tests for that purpose :)